src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketsTelemetry.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Tracing;
+
+namespace System.Net.Sockets
+{
+    [EventSource(Name = "System.Net.Sockets")]
+    internal sealed class SocketsTelemetry : EventSource
+    {
+        public static readonly SocketsTelemetry Log = new SocketsTelemetry();
+
+        [NonEvent]
+        public void ConnectStart(Internals.SocketAddress address)
+        {
+            if (IsEnabled())
+            {
+                ConnectStart(address.ToString());
+            }
+        }
+
+        [NonEvent]
+        public void ConnectStart(EndPoint address)
+        {
+            if (IsEnabled())
+            {
+                var addressString = address.ToString();
+                if (addressString != null)
+                {
+                    ConnectStart(addressString);
+                }
```

Comment:
This looks _super_ corner-casey, but if I had counters enabled I would expect an entry for every connect -- instead of not logging on null, can we log e.g. `""` or `"<null>"`?

Reply:
Done